compiler/parser.nim

```diff
@@ -744,8 +761,14 @@ proc primarySuffix(p: var TParser, r: PNode, baseIndent: int): PNode =
       # progress guaranteed
       somePar()
       result = namedParams(p, result, nkCurlyExpr, tkCurlyRi)
-    of tkSymbol, tkAccent, tkIntLit..tkCharLit, tkNil, tkCast, tkAddr, tkType:
-      if p.inPragma == 0:
+    of tkSymbol, tkAccent, tkIntLit..tkCharLit, tkNil, tkCast,
+       tkTypeClasses - {tkRef, tkPtr}:
+        # XXX: In type sections we allow the free application of the
+        # command syntax, with the exception of expressions such as
+        # `foo ref` or `foo ptr`. Unfortunately, these two are also
+        # used as infix operators for the memory regions feature and
```

Comment:
Removing the infix version of these is fine with me if it simplifies the logic here.

Reply:
Please answer this question for me. :-)

Reply:
Let's do this later. I'd like to work on other more pressing issues.

Reply:
If it's just a matter of removing the parsing logic around the infix operator, perhaps it's a very small patch, but I haven't looked into it yet.